### ISSUE TITLE
Link the text inside a shape to the shape

### ISSUE THREAD

**Issue by pshihn** [excalidraw/excalidraw#1010]
It can get a bit annoying when trying to move a block but it doesn't move the text inside it. Not a big issue, but I have forgotten to rubber-band (or command click) to move them together way too many times now :(

(I think there's a separate issue about linking arrows to shapes which is related)

**Comment by dwelle** [comment]
Related to (and possibly blocked by) https://github.com/excalidraw/excalidraw/issues/294. Let's keep this open, but as for how to implement this, let's move there because it'll require more meta discussion.

**Comment by ReneCode** [comment]
What's about the idea to move also all elements that are completely inside the moving element.
So any element will be moved when its surrounding element is moved.

That behaviour will not be limited to text inside box, but working for all element types.

![moveTextWithBox](https://user-images.githubusercontent.com/1595098/87094662-86e36880-c23f-11ea-9893-367d9298309e.gif)

**Comment by xixixao** [comment]
I think this issue is nicely and more generically addressed by groups. @ReneCode give it a try (select multiple elements, cmd+option+g on mac).

**Comment by ReneCode** [comment]
@xixixao yes, you are right -looks better using group. Because on resize the text-element will be resized too.

May be thinking about an "automatic-grouping" approach would help to make the workflow smoother, like:
* draw rectangle
* draw text (starting point is **inside** the rectangle => both are grouped)
* move rectangle
=> rectangle **and** text are moved.

**Comment by xixixao** [comment]
Grouping text to the shape when the text is created (via double click or text tool) **while** the shape is selected might work. It should still not autogroup if the shape is not selected and I'm adding text inside of it.

(then again any of this could be too much magic and stirring away from "just whiteboard" approach)

**Comment by ReneCode** [comment]
I've tried out to make auto-grouping when the new text is inside an element.
[try here](https://excalidraw.relang.vercel.app/)

It works fine - if you leave the size of the element. But on resizing the element the text is resized too. That results a bad looking result when in a diagram with several elements with text the text size is different. 

may be this approach is not ideal.

**Comment by xixixao** [comment]
@ReneCode you need to double click the group and then only resize the rectangle (so I don't think this is a big problem).

I wonder how @vjeux likes this version.

**Comment by dwelle** [comment]
Using grouping to solve this case came across my mind as well, but @ReneCode has a valid point. As an interim solution maybe?

**Comment by vjeux** [comment]
I feel like it achieves the behavior we want: when you move the shape it moves the text around. Now there are some unfortunate side effects:
- Scaling the shape: now it's just in one axis and not the other. It also scales the text which is unexpected.
- Changing the text is harder. You need to quadruple click. Inner selection is a bit more weird.

I feel like the downsides are greater than the wins. I feel like once the arrow linking is implemented, we can use the same link in order to move things around and it wouldn't have those drawbacks.

**Comment by xixixao** [comment]
@vjeux Do you imagine we'll allow binding text to rectangles? I think this runs into a problem if we also want to allow binding of arrows to text, see bottom of my initial PR comment: #1899 .

**Comment by vjeux** [comment]
Interesting. I didn't think you'd be able to bind arrows to text fwiw.

**Comment by ttjmartinez** [comment]
I like the way it is done in yED, when you drag some object <x> into some other object <y>, if you have <shift> pressed at the time release the button, <x> becomes a component within <y>, and from that time on they move together whenever <y> moves. You can still move <x> inside <y>, but if you try to drag <x> outside of <y>, then <y> needs to grow in order to keep containing it. In order to detach <x> from <y>, grab it, then press <shit> move it out of <y> and release the button.

**Comment by fqf2009** [comment]
Text actually has a border which is just transparent. It will show up when you move the end-point of a line over the text border. Actually the line will be attached to text border when you release the mouse at this moment. So, why not just add a style to text border to show it? This will be quite convenient and useful.

**Comment by Frazer** [comment]
Perhaps it could work just like when you make an arrow, that connects to a shape.  As you get close, a drop zone appears, and if you drop there it is connected.

**Comment by bzadon** [comment]
+1 vote for this feature. Perhaps "Auto-Group" is a preference instead of changing the default? Keeps both camps happy.

**Comment by talperets** [comment]
Hey, I don't know if it's still relevant but if it is, I would love to work on that one
